pandas/core/nanops.py

```diff
@@ -344,12 +344,22 @@ def nanmedian(values, axis=None, skipna=True):
 
     values, mask, dtype, dtype_max = _get_values(values, skipna)
 
+    if not skipna:
```

Comment:
The tests compare nanmedian to numpy's median function. If this special case were not there, this function would return a non-null result for input with null values and skipna turned off. Here is a debug log demonstrating this: https://gist.github.com/rohanp/1082ad5a5e199f6b1cb8ade965c4519f